### ISSUE TITLE
Group block content placeholder is misaligned

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#26379]
**Describe the bug**
When adding a group block that is wide or full width, the content placeholder is left aligned.
But when you add content that replaces the placeholder, the content is in the middle of the block.

**To reproduce**
1. Add 3 group blocks, one with the default with, one wide, and one full width.
2. See the difference of the position of the placeholder in the 3 versions.

**Expected behavior**
I expect the placeholder to be in the same position no matter the width.

**Screenshots**
![group-block](https://user-images.githubusercontent.com/7422055/96842885-e7dd2800-144d-11eb-9941-80a86d30aea5.png)

**Editor version (please complete the following information):**
- WordPress version: 5.6-beta1-49267
-With and without Gutenberg master, with and without Gutenberg 9.2

**Desktop (please complete the following information):**
 - OS: Windows 10, Chrome version 86.0.4240.111

**Comment by carolinan** [comment]
I have confirmed that this is still an issue. 

@jasmussen The appender seems to be left aligned due to these margins, which are not changed depending on the block width?

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-list-appender/style.scss#L9
```
.block-editor-block-list__block .block-list-appender:first-child {
    margin-left: 0;
}

.block-editor-block-list__block .block-list-appender {
    ...
    margin: 0 0 0 8px;
}
```

**Comment by jasmussen** [comment]
I can reproduce the issue in TT1 Blocks:

<img width="1534" alt="Screenshot 2021-02-22 at 11 37 04" src="https://user-images.githubusercontent.com/1204802/108696791-4c92b700-7502-11eb-9797-f514da268460.png">

I cannot reproduce it in my own theme:

<img width="1689" alt="Screenshot 2021-02-22 at 11 35 17" src="https://user-images.githubusercontent.com/1204802/108696809-51f00180-7502-11eb-8181-ca03c6ebc76f.png">

I'm still investigating what has caused this, and will look at a fix.

> I expect the placeholder to be in the same position no matter the width.

I'm not sure I agree — I would expect the placeholder to be the same width of the block itself. I know paragraphs inside are in a centered column — but the placeholder state is related to the container block and should help indicate that this group is actually wide or full-wide.

**Comment by jasmussen** [comment]
So there are two fixes to this one. 

There's this one:

```
.block-list-appender.wp-block {
    max-width: none;
}
```

That makes the appender take up the full available space by explicitly unsetting the block max-width set by themes like TT1-blocks.

<img width="1556" alt="Screenshot 2021-02-22 at 11 42 43" src="https://user-images.githubusercontent.com/1204802/108697397-17d32f80-7503-11eb-8af4-fad028a5e614.png">

That assumes that the appender _should_ be full-width, which I lean towards because it helps indicate the state of the empty group block, which seems useful to me.

---

The other fix could be this one:

```
.block-editor-block-list__block .block-list-appender:first-child {
    margin-left: auto;
}
```

which results in this:

<img width="1261" alt="Screenshot 2021-02-22 at 11 45 15" src="https://user-images.githubusercontent.com/1204802/108697709-7c8e8a00-7503-11eb-9ac9-544c4222abd9.png">

While that's definitely better than the current staircase effect, it doesn't feel that useful to me. An empty/unset group block that's aligned wide or full-wide has the opportunity to communicate its properties, and wide and fullwide feel worth communicating. 

What do you think?

**Comment by carolinan** [comment]
I would prefer if the appender was the same width as the block. It is easier to see the size that way.
If the block is wide, the appender should be wide but centered, not positioned to the left.

The first Screenshot was of Twenty Twenty-One.
Here is another shot of Twenty Twenty and Gutenberg Version 10.0.1. I am seeing the same result with the current master.
The block editor, -not site editor- in Chrome Version 88.0.4324.190, Windows 10.

![image](https://user-images.githubusercontent.com/7422055/108829331-55e65700-75c8-11eb-84d6-24bc0ba18b85.png)

**Comment by carolinan** [comment]
Seedlet

![image](https://user-images.githubusercontent.com/7422055/108830351-8b3f7480-75c9-11eb-9cd8-fd5783ac0da9.png)

**Comment by jasmussen** [comment]
Fixed in #29290 29290
